paddle/lib/sgd_optimizer.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __PADDLE_SGD_OPTIMIZER_H__
+#define __PADDLE_SGD_OPTIMIZER_H__
+
+#include "parameter_optimizer.h"
+#include "training_ops.h"
+
+class SGDOptimizer : public ParameterOptimizer {
+  public:
+  /*! \brief call the applySGD for example  */
+  void update(Tensor<T> &parameter,
+              const Tensor<T> &gradient,
+              double learning_rate) {
+    applyGradientDescent(parameter, gradient, learning_rate);
+  }
+};
+
+class MomentumOptimizer : public ParameterOptimizer {
+
+public:
+  /*! \brief call the applyXX for example  */
+  void update(Tensor<T> &parameter,
```

Comment:
This function has a different signature than the parent pure virtual function `update`, consider change the parent `update` function signature.